buildlib/azure-pipelines-pr.yml

```diff
@@ -47,6 +47,29 @@ stages:
                  echo "##vso[task.complete result=Failed;]"
               fi
             condition: eq(variables['Build.Reason'], 'PullRequest')
+  - stage: Format
+    jobs:
+      # Check that the code is formatted according to the code style guidelines
+      - job: format
+        displayName: format code
+        container: fedora
+        steps:
+          - checkout: self
+            clean: true
+
+          - bash: |
+              git-clang-format --diff remotes/origin/${SYSTEM_PULLREQUEST_TARGETBRANCH} ${BUILD_SOURCEVERSION} > /tmp/patch
+              if [ "`cat /tmp/patch`" = "no modified files to format" ]; then
+                exit
+              fi
+              git apply /tmp/patch
```

Comment:
can't we check if patch is empty? or check it's diff stat?

Reply:
Unfortunately, the only way to distinguish between a "well-formatted" state and one that needs fixes, is to check whether the output of `git-clang-format` is exactly the string `"no modified files to format"`. In case there is nothing to format, it doesn't produce empty output or any other exit code. And we can't pass that text in pipe to `git apply` - that would yield an error.